src/components/tree/tree-item.ts

```diff
@@ -0,0 +1,541 @@
+import { IgcTreeSelectionType } from './tree.common.js';
+import { watch } from '../common/decorators';
+import { IgcTreeSelectionService } from './tree.selection.js';
+import { IgcTreeNavigationService } from './tree.navigation.js';
+import { themes } from '../../theming';
+
+/**
+ * The tree-item component represents a child item of the tree component or another tree item.
+ *
+ * @element igc-tree-item
+ *
+ * @slot - Renders nested tree-item component.
+ * @slot label - Renders the tree item container.
+ * @slot indicator - Renders the expand indicator container.
+ * @slot loading - Renders the tree item loading indicator container.
+ * @slot indentation - Renders the container (by default the space) before the tree item.
+ */
+@themes({ bootstrap, fluent, indigo })
+export default class IgcTreeItemComponent extends LitElement {
+  /** @private */
+  public static tagName = 'igc-tree-item';
+  /** @private */
+  public static override styles = styles;
+
+  private tabbableEl?: HTMLElement[];
+  private focusedProgrammatically = false;
+
+  /** A reference to the tree the item is a part of. */
+  public tree?: IgcTreeComponent;
+  /** The parent item of the current tree item (if any) */
+  public parent: IgcTreeItemComponent | null = null;
+
+  /** @private */
+  public init = false;
+
+  @queryAssignedElements({ slot: 'label', flatten: true })
+  private contentList!: Array<HTMLElement>;
+
+  /** @private */
+  @query('#wrapper')
+  public wrapper!: HTMLElement;
+
+  @state()
+  private isFocused = false;
+
+  /** @private */
+  @state()
+  public hasChildren = false;
+
+  /** @private */
+  @state()
+  public size = 'large';
+
+  /** @private */
+  @state()
+  public override dir = 'ltr';
+
+  /** The depth of the node, relative to the root. */
+  @state()
+  public level = 0;
+
+  /** @private */
+  @state()
+  public indeterminate = false;
+
+  /** @private */
+  @state()
+  public selection: IgcTreeSelectionType = IgcTreeSelectionType.None;
+
+  /** The tree item label. */
+  @property()
+  public label = '';
+
+  /** The tree item expansion state. */
+  @property({ reflect: true, type: Boolean })
+  public expanded = false;
+
+  /** Marks the item as the tree's active item. */
+  @property({ reflect: true, type: Boolean })
+  public active = false;
+
+  /** Get/Set whether the tree item is disabled. Disabled items are ignored for user interactions. */
+  @property({ reflect: true, type: Boolean })
+  public disabled = false;
+
+  /** The tree item selection state. */
+  @property({ reflect: true, type: Boolean })
+  public selected = false;
+
+  /** To be used for load-on-demand scenarios in order to specify whether the item is loading data. */
+  @property({ reflect: true, type: Boolean })
+  public loading = false;
+
+  /**
+   * The value entry that the tree item is visualizing. Required for searching through items.
+   * @type any
+   */
+  @property({ attribute: true })
+  public value: any = undefined;
```

Comment:
```suggestion
  @property()
  public value: string = '';
```

Type this as a string.